src/UseCases/CancelDonation/CancelDonationUseCase.php

```diff
@@ -22,12 +23,30 @@ public function __construct( DonationRepository $donationRepository, TemplateBas
 	}
 
 	public function cancelDonation( CancelDonationRequest $cancellationRequest ): CancelDonationResponse {
+		try {
+			$donation = $this->donationRepository->getDonationById( $cancellationRequest->getDonationId() );
+		}
+		catch ( GetDonationException $ex ) {
+			return new CancelDonationResponse( $cancellationRequest->getDonationId(), false );
+		}
+
+		if ( $donation === null ) {
+			return new CancelDonationResponse( $cancellationRequest->getDonationId(), false );
+		}
+
+		try {
+			$donation->cancel();
+		}
+		catch ( \RuntimeException $ex ) {
+			return new CancelDonationResponse( $cancellationRequest->getDonationId(), true );
```

Comment:
Seems like someone wrote production code that was not needed to make the test pass...